collectors/REFERENCE.md

```diff
@@ -40,7 +41,41 @@ There are three types of plugins:
     independent processes. They communicate with the daemon via pipes.
 -   **Plugin orchestrators**, which are external plugins that instead support a number of **modules**. Modules are a
     type of collector. We have a few plugin orchestrators available for those who want to develop their own collectors,
-    but focus most of our efforts on the [Go plugin](go.d.plugin/).
+    but focus most of our efforts on the [Go plugin](go.d.plugin/README.md).
+
+## Troubleshoot a collector
+
+First, naviagate to your plugins directory, which is usually at `/usr/libexec/netdata/plugins.d/`. If that's not
+the case on your system, open `netdata.conf` and look for the setting `plugins directory`. Once you're in the plugins
+directory, switch to the `netdata` user.
+
+```bash
+cd /usr/libexec/netdata/plugins.d/
+sudo su -s /bin/bash netdata
+```
+
+The next step is based on the collector's orchestrator. You can figure out which orchestrator the collector uses either
+by viewing the [collectors list](COLLECTORS.md) and referencing the _configuration file_ field. For example, if that
+field contains `go.d`, that collector uses the Go orchestrator.
+
+```bash
+# Go orchestrator (go.d.plugin)
+./go.d.plugin -d -m <MODULE_NAME>
+
+# Python orchestrator (python.d.plugin)
+./python.d.plugin <MODULE_NAME> debug trace
+
+# Node orchestrator (node.d.plugin)
+./node.d.plugin debug 1 <MODULE_NAME>
+
+# Bash orchestrator (bash.d.plugin)
+./charts.d.plugin debug 1 <MODULE_NAME>
```

Comment:
not sure that thing have per module debug mode, same for `node.d.plugin` 😄

Reply:
it seems bash has no (at least `debug 1 name` doesnt work)

```
st@DEB10-1:/opt/netdata/usr/libexec/netdata/plugins.d$ sudo su
root@DEB10-1:/opt/netdata/usr/libexec/netdata/plugins.d# ./charts.d.plugin debug 1 nut
2020-02-10 09:58:29: charts.d: INFO: main: started from './charts.d.plugin' with options: debug 1 nut
2020-02-10 09:58:29: charts.d: INFO: main: Configuration file '/opt/netdata/usr/lib/netdata/conf.d/charts.d.conf' loaded.
2020-02-10 09:58:29: charts.d: WARNING: main: Configuration file '/opt/netdata/etc/netdata/charts.d.conf' not found.
2020-02-10 09:58:29: charts.d: INFO: apache: is disabled. Add a line with apache=force in '/opt/netdata/etc/netdata/charts.d.conf' to enable it (or remove the line that disables it).
2020-02-10 09:58:29: charts.d: DEBUG: ap: is enabled for auto-detection.
2020-02-10 09:58:29: charts.d: DEBUG: apcupsd: is enabled for auto-detection.
```
___

`node.d.plugin` HAS per module debug mode

Reply:
According to [the docs](https://docs.netdata.cloud/collectors/charts.d.plugin/#debugging-your-collectors), `charts.d.plugin` _should_ be able to use the syntax, but you're right that it doesn't actually work that way. No matter what module you put in it returns output from all modules.

Reply:
ok, i am wrong, it works, but 

- it support only 1 module in debug mode (`./charts.d.plugin debug 1 X`)
- it **loads all** modules (configs, runs checks for all of them)
- and only after it keeps running the one that is in debug

Reply:
that is debug code

https://github.com/netdata/netdata/blob/d3fa405141a8d2e6689c2dedb4bedb7cf9120044/collectors/charts.d.plugin/charts.d.plugin.in#L546-L558

load all modules, run check method for all of them code before it

https://github.com/netdata/netdata/blob/d3fa405141a8d2e6689c2dedb4bedb7cf9120044/collectors/charts.d.plugin/charts.d.plugin.in#L489-L534